pandas/tests/groupby/test_any_all.py

```diff
@@ -68,3 +69,91 @@ def test_bool_aggs_dup_column_labels(bool_agg_func):
 
     expected = df
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("bool_agg_func", ["any", "all"])
+@pytest.mark.parametrize("skipna", [True, False])
+@pytest.mark.parametrize(
+    # expected_data indexed as [[skipna=False/any, skipna=False/all],
+    #                           [skipna=True/any, skipna=True/all]]
+    "data,expected_data",
+    [
+        ([False, False, False], [[False, False], [False, False]]),
+        ([True, True, True], [[True, True], [True, True]]),
+        ([pd.NA, pd.NA, pd.NA], [[pd.NA, pd.NA], [False, True]]),
+        ([False, pd.NA, False], [[pd.NA, False], [False, False]]),
+        ([True, pd.NA, True], [[True, pd.NA], [True, True]]),
+        ([True, pd.NA, False], [[True, False], [True, False]]),
+    ],
+)
+def test_masked_kleene_logic(bool_agg_func, data, expected_data, skipna):
+    # GH#37506
+    df = DataFrame(data, dtype="boolean")
+    expected = DataFrame(
+        [expected_data[skipna][bool_agg_func == "all"]], dtype="boolean", index=[1]
+    )
+
+    result = df.groupby([1, 1, 1]).agg(bool_agg_func, skipna=skipna)
+    tm.assert_frame_equal(result, expected)
+
+    # The expected result we compared to should match aggregating on the whole
+    # series
+    result = getattr(df[0], bool_agg_func)(skipna=skipna)
```

Comment:
this check is really hard to parse.
what exactly are you checking? can you make this much simpler

Reply:
What this is essentially doing is `series.any(skipna=skipna)`, but then with `getattr` because it's generic for any/all. Do you have a concrete suggestion to change? (or to clarify the comment?)

Reply:
there is a complicated expexcted then a really complicated assert. This needs to be greatly simplified. It is impossible to grok with all of this logic. my suggesetion is to break this up.

Reply:
To be very concrete, you mean breaking it up into 2 lines, like

```
series = df[0]
getattr(series, method)(skipna=skipna)
```

?